### PR TITLE
delete todo

### PR DESCRIPTION
**Pull request by fork-boy**
## Description

Delete API for todo task
DELETE /todos/:id

## Database schema changes

Updated Schema:
{
title: { type: String, required: true },
isActive: { type: Boolean, default: true },
},
{
collection: "todos",
timestamps: {
createdAt: "createdAt",
updatedAt: "updatedAt",
},
}

### Automated test cases added

- should return 404 if id is empty.
- should return 204 if id exists and success delete

### Manual test cases run

- None

### CODE REVIEW

File: src/controllers/todo-controller.ts [hunk 1]
```diff
@@ -51,4 +53,23 @@ export class TodoController extends BaseController {
...
+
+    const todo = await this.appContext.todoRepository.update(
+      { _id: id, isActive: true },
+      { $set: { isActive: false } }
+    );
+
+    if (todo?._id) {
+      res.status(204).send();
+    } else {
+      res.status(404).send();
```
Comment (jjalan) [inline src/controllers/todo-controller.ts]: We should define custom error message. I think we already for others in errors folder

**Comment by vindru** [comment]
@agentraghav  I am not seeing the schema changes that you mentioned in the PR description in file changes

File: test/spec/todo/todo.spec.ts [hunk 2]
```diff
@@ -59,3 +59,16 @@ describe('POST /todos', () => {
...
+describe('DELETE /todos/:id', () => {
+  it('should return 204 if todo exists else 404', async () => {
+    let todo = await testAppContext.todoRepository.save(
+      new TodoItem({ title: 'TODO_TEMPORARY' })
+    );
+    const res1 = await chai.request(expressApp).delete(`/todos/${todo._id}`);
+    expect(res1).to.have.status(204);
+
+    const res2 = await chai.request(expressApp).delete(`/todos/${todo._id}`);
+    expect(res2).to.have.status(404);
```
Comment (vindru) [inline test/spec/todo/todo.spec.ts:81]: Please also add test for error message that jai asked to add above

File: src/controllers/todo-controller.ts (hunk 1 above)
Comment (fork-boy) [reply src/controllers/todo-controller.ts]: ok

File: test/spec/todo/todo.spec.ts (hunk 2 above)
Comment (fork-boy) [reply test/spec/todo/todo.spec.ts:81]: ok

File: src/controllers/todo-controller.ts [hunk 3]
```diff
@@ -51,4 +53,26 @@ export class TodoController extends BaseController {
...
+  ) => {
+    const { id } = req.params;
+
+    const todo = await this.appContext.todoRepository.update(
+      { _id: id, isActive: true },
+      { $set: { isActive: false } }
+    );
+
+    if (todo?._id) {
+      res.status(200).json(todo.serialize());
```
Comment (vindru) [inline src/controllers/todo-controller.ts]: response 204 was correct here with no response. 200 is for the get requests only.

**Pull request merged**